### ISSUE TITLE
[question] Focal Loss alpha parameter intuition

### ISSUE THREAD

**Issue by v-shmyhlo** [fizyr/keras-retinanet#540]
Can anybody please explain to me the purpose of `alpha` parameter in Focal Loss?
I have built my own [implementation](https://github.com/v-shmyhlo/retinanet-tensorflow) of RetinaNet from scratch using TensorFlow and I am struggling with fitting it for almost 2 months. The loss converges to some high value and the network predicts everything as a background.
I've tried a couple of toy experiments to understand what is happening, but it didn't give me any insights.
What I can't understand is what is the intuition behind `0.25` value of `alpha` parameter proposed in the paper? As I understand, the purpose of `alpha` is to balance the foreground and background samples, but from the definition of Focal Loss it looks like it downweights foreground samples and upweights background samples, which doesn't make sense for me.
Can somebody explain me this part?

**Comment by hgaiser** [comment]
I vaguely remember having a small email conversation with the author of the original paper about this. He said that anchors are likely already classified as background, so they exhibit a lower loss than the foreground samples. That's why foreground is down weighed.

For more theoretical questions / discussions, please join the Slack channel. That way we can keep the issues for implementation issues.

**Comment by czero69** [comment]
Did you maybe use **adam** as an optimization method? I had a similar issue similar to retinaNet detector structure when I was trying adam (everything was detected only as background) but moving to sgd+momentum makes the detector catching the task. So if you are using adam, I would suggest instead of adam to try using sgd+momentum. I am now experimenting with a detector which shares a lot of ideas from retinaNet, and it is more and more intuitive for me why adam fails in my case. Still, the loss is dependent on a few detections. Adam catches very quick on predicting everything as a background but is unable to start to progressing towards detecting objects. Sgd+momentum is training slower, but can break through this barrier of detecting objects and not the only background. I use different loss than focal loss (based on dlib MMOD loss, which still is more similar to cross entropy loss than focal loss), and trying focal mmod loss is on the list.

**Comment by hgaiser** [comment]
Yeah I would like to see how keras-retinanet performs using SGD, but I don't have the time to try it out :( If anyone feels like it, be my guest.

**Comment by HyunjiEllenPak** [comment]
@hgaiser 
 I don't understand why optimal alpha is set under 0.5 from your explanations(="He said that anchors are likely already classified as background, so they exhibit a lower loss than the foreground samples. That's why foreground is down weighed").

If alpha is 0.25, a predicted box that is included in foreground region would get the loss L=-0.25 (1-p)^{gamma}log(p). 
And a predicted box that is included in background would get the loss L=-0.75 p^{gamma} log (1-p) 
It seems the background box weights more than foreground to the loss.

Could you explain more?

p.s. I can't find any specific explanation or some reasons why optimal alpha is under 0.5 in focal loss..
